lightning/src/routing/scoring.rs

```diff
@@ -713,8 +725,8 @@ impl<G: Deref<Target = NetworkGraph<L>>, L: Deref, T: Time> Score for Probabilis
 	fn channel_penalty_msat(
 		&self, short_channel_id: u64, source: &NodeId, target: &NodeId, usage: ChannelUsage
 	) -> u64 {
-		if self.params.banned_nodes.contains(source) || self.params.banned_nodes.contains(target) {
-			return u64::max_value();
+		if let Some(penalty) = self.params.manual_node_penalties.get(target) {
```

Comment:
I see your point that a penalty for the source would be considered a bit earlier. However, I chose the target here since it generally seems more in line with the behavior I would expect, and in particular it allows us to penalize/prefer specific single-hop paths. This could come handy when a user would like to use this mechanism to always choose a particular channel when routing to a neighbor node.

Reply:
> Wait, can it? If you're paying a given peer, this API doesn't let you pick a channel to that peer, you can punish that peer, but I'm not sure what that does?

Well I assume when you have multiple channels open you can set a high/low penalty to prefer/avoid a specific one over the others for example? I'd think this could for example get used for simple balancing purposes, i.e., set higher penalties the more a specific channel is imbalanced. A nice side-effect would be that you could always still fall back on a penalized outgoing channel, as long as you not set it outright to `u64::max_value()`.

Reply:
Yes, this gives us the possibility to essentially prioritize first hops. I agree that a more fine-grained prioritization on a per-channel basis could be an improvement, but I think this can be a good first step.